File: tessoku/B03 - Supermarket 1/B03.ts
const mainB03 = (input: string): void => {
    const lines: string[] = input.trim().split('\n');
    const N: number = Number(lines[0]);
    const prices: number[] = lines[1].split(' ').map(Number);
    let result: string = 'No';

    for (let i: number = 0; i < N - 2; i++) {
        for (let j: number = i + 1; j < N - 1; j++) {
            for (let k: number = j + 1; k < N; k++) {
                if (prices[i] + prices[j] + prices[k] === 1000) {
                    result = 'Yes';
                    break;
                }
            }
        }
    }

    console.log(result);
}

mainB03(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: tessoku/B03 - Supermarket 1/B03.cs
using System;
using System.Linq;

class Program
{
   static void Main()
   {
       int N = int.Parse(Console.ReadLine());

       int[] prices = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
       
       string result = "No";

       for (int i = 0; i < N - 2; i++)
       {
           for (int j = i + 1; j < N - 1; j++)
           {
               for (int k = j + 1; k < N; k++)
               {
                   if (prices[i] + prices[j] + prices[k] == 1000)
                   {
                       result = "Yes";
                       goto End;
                   }
               }
           }
       }

       End:
       Console.WriteLine(result);
   }
}